core/src/test/scala/unit/kafka/network/RequestChannelTest.scala

```diff
@@ -255,6 +258,141 @@ class RequestChannelTest {
     assertTrue(capturedValue.error().equals(Errors.NOT_CONTROLLER))
   }
 
+  @Test
+  def testMetricsRequestSizeBucket(): Unit = {
+    val apis = Seq(ApiKeys.FETCH, ApiKeys.PRODUCE)
+    var props = new Properties()
+    props.put(KafkaConfig.ZkConnectProp, "127.0.0.1:2181")
+    props.put(KafkaConfig.RequestMetricsSizeBucketsProp, "0, 10, 20, 200")
+    var config = KafkaConfig.fromProps(props)
+    var metrics = new Metrics(apis, config)
+    val fetchMetricsNameMap = metrics.consumerFetchRequestSizeMetricNameMap
+    assertEquals(4, fetchMetricsNameMap.size)
+    assertTrue(fetchMetricsNameMap.contains(0))
+    assertTrue(fetchMetricsNameMap.contains(10))
+    assertTrue(fetchMetricsNameMap.contains(20))
+    assertTrue(fetchMetricsNameMap.contains(200))
+    val flattenFetchMetricNames = metrics.getConsumerFetchRequestSizeMetricNames
+    assertEquals(4, flattenFetchMetricNames.size)
+    assertEquals("FetchConsumer0To10Mb", fetchMetricsNameMap(0))
+    assertTrue(flattenFetchMetricNames.contains("FetchConsumer0To10Mb"))
+    assertEquals("FetchConsumer10To20Mb", fetchMetricsNameMap(10))
+    assertTrue(flattenFetchMetricNames.contains("FetchConsumer10To20Mb"))
+    assertEquals("FetchConsumer20To200Mb", fetchMetricsNameMap(20))
+    assertTrue(flattenFetchMetricNames.contains("FetchConsumer20To200Mb"))
+    assertEquals("FetchConsumer200MbGreater", fetchMetricsNameMap(200))
+    assertTrue(flattenFetchMetricNames.contains("FetchConsumer200MbGreater"))
+
+    val produceMetricsNameMaps = metrics.produceRequestAcksSizeMetricNameMap
+    assertEquals(3, produceMetricsNameMaps.size)
+
+    assertTrue(produceMetricsNameMaps.contains(0))
+    assertTrue(produceMetricsNameMaps.contains(1))
+    assertTrue(produceMetricsNameMaps.contains(-1))
+    val flattenProduceMetricNames = metrics.getProduceRequestAcksSizeMetricNames
+    assertEquals(12, flattenProduceMetricNames.size)
+    var i = 0
```

Comment:
Minor: This line can be removed since the next line has an implicit declaration of the variable i.

Reply:
Removed